src/main/java/vazkii/botania/api/recipe/IManaInfusionRecipe.java

```diff
@@ -27,6 +27,15 @@
 
 	boolean matches(ItemStack stack);
 
+	@Nonnull
+	default ItemStack getRecipeOutputDynamic(@Nonnull ItemStack input) {
```

Comment:
Just call it `getRecipeOutput`, just like the elven trade recipe has one method with no params and one with a list of stacks.